server/schedule/operator/operator.go

```diff
@@ -757,9 +819,33 @@ func removePeerSteps(cluster Cluster, region *core.RegionInfo, storeID uint64, f
 	return
 }
 
+// findNoRejectLeader finds the first store without RejectLeader label.
+func findNoRejectLeader(storeIDs []uint64, cluster Cluster) (int, uint64) {
+	for i, id := range storeIDs {
+		store := cluster.GetStore(id)
+		if store != nil {
+			if !cluster.CheckLabelProperty(opt.RejectLeader, store.GetLabels()) {
+				return i, id
+			}
+		} else {
+			log.Debug("nil store", zap.Uint64("store id", id))
+		}
+	}
+	return -1, 0
+}
+
+// transferLeaderToAnySteps returns the first suitable store to become region leader,
+func transferLeaderToAnySteps(leaderID uint64, storeIDs []uint64, cluster Cluster) (OpKind, []OpStep) {
```

Comment:
If we regard this function as a black box, we should not assume it will only return a `TransferLeader` step if succeeded.

What's more, I'd like to make `*Steps` functions have similar returns. Now there are 3 such functions that have different returns:

+ `transferLeaderToAnySteps`(this function): no error.
+ `CreateAddPeerSteps` and `CreateAddLightPeerSteps`: no OpKind and error.

I will add an error for this function, because it may fail.
The other two functions are exported, so I didn't touch them.